### ISSUE TITLE
nodepools.kube.cloud.ovh.com enhancements

### ISSUE THREAD

**Issue by rverchere** [ovh/public-cloud-roadmap#411]
Hello,

When editing the `nodepools.kube.cloud.ovh.com` CRD on my cluster, I can:
- Edit the number of nodes : the nodes are correctly updated
- Edit the antiAffinity : the edition is possible but not reflected on the backend side (API/Manager)
- Edit maxNodes : decreasing the value less than the current nodes is OK. IIncrease is possible, but no behaviour (and I can set to A LOT of max nodes)

I suggest to validate some CRDs specs of `nodepools.kube.cloud.ovh.com` to be sure that the CRD values are correct and reflecting the current state of the cluster.

I see at least these specs to be enhanced:
- antiAffinity
- autoscale
- maxNodes (increase check)

**Comment by mhurtrel** [comment]
Hello @rverchere 
antiaffinity is set at nodepool creation, but cannot be edited at a later stage (due to the way it is leveraging openstack and our supply management). T

I understand that we should add some validation/error to be more explicit about this and also make sure that max nodes is between current node count and 100 (our limit per node pool).

I will ask the team to evaluate this, thanks for the input !

**Comment by thcdrt** [comment]
Hello, there is the same issue with flavor.

**Comment by XavierDuthil** [comment]
Hello, after some discussions we determined that a validatingwebhook would not be be an ideal solution, as it would introduce scheduling issues.

At some point we will be able to use the new CRD [Validation rules](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules) feature, but as it is only available in beta starting from K8S v1.25, and we still support some previous versions, we will refrain from switching to it in order to keep a consistency between our versions.

However in the meantime, a quick fix will be implemented to our nodepool operator to execute these validations after a CR was updated, and fix any inconsistency found in the specs. This will not prevent you from updating the CR, but it will ensure that it gets back to a valid state.
